cpp/ql/src/Security/CWE/CWE-676/DangerousUseOfGets.ql

```diff
@@ -0,0 +1,18 @@
+/**
+ * @name Use of dangerous function 'gets'
+ * @description The standard library 'gets' function is dangerous and should not be used.
+ * @kind problem
+ * @problem.severity error
+ * @precision very-high
+ * @id cpp/potentially-dangerous-function
```

Comment:
This query needs its own query ID. Even though only `gets` is matched right now, I suggest making the ID slightly more general so it can also accommodate `sprintf("%s", ...)` in the future if we want that. Maybe `@id cpp/unbounded-read-from-file` or something like that. The file name should match whatever we decide on for query id.

Reply:
Ah, missed the query ID!  Fixed now, and I've changed the name to be a bit more general than just 'gets'.